tests/lib/state_test.py

```diff
@@ -697,3 +698,35 @@ def _route_sort_key(route):
     return (route.get(Route.TABLE_ID, Route.USE_DEFAULT_ROUTE_TABLE),
             route.get(Route.NEXT_HOP_INTERFACE, ''),
             route.get(Route.DESTINATION, ''))
+
+
+def test_merge_config_dns():
```

Comment:
Please use a class to group these tests.
Regarding the test name: Please express what it is exactly checking, the current name is too generic.
Maybe: `test_merge_dns_empty_state_with_non_empty_state`